power_grid_model_c/power_grid_model/include/power_grid_model/all_components.hpp

```diff
@@ -27,9 +27,56 @@
 #include "component/voltage_sensor.hpp"
 
 namespace power_grid_model {
+namespace detail {
+template <typename List, typename T, typename... NeedsTypes>
+concept dependent_type_check = !IsInList<T, List>::value || (IsInList<NeedsTypes, List>::value && ...);
```

Comment:
I was checking a different condition previously. It now seems to be possible to move these checks to MainModelType. Ill do so now.